### ISSUE TITLE
PDF viewer does not update page count on reload

### ISSUE THREAD

**Issue by allispaul** [emacs-eaf/eaf-pdf-viewer#48]
**Describe the bug**
If the number of pages in the PDF changes and the viewer reloads the PDF, it will reload it with the old number of pages, either making pages at the end inaccessible or showing empty blank space and throwing an error.

**To Reproduce**
Using AucTeX and the LaTeX package `lipsum`:
1. Create a new TeX file called containing the following text:
```\documentclass{article}
\usepackage{lipsum}
\begin{document}
\lipsum[1]
\end{document}
```
2. Build the file and open the PDF in eaf-pdf-viewer. It should be 1 page long.
3. Change the line `lipsum[1]` to `lipsum[1-10]`.
4. The PDF auto-reloads but only shows the first of 2 pages.
5. Close and reopen the PDF in eaf-pdf-viewer. It should now show both pages.
6. Now change the line `lipsum[1-10]` back to `lipsum[1]`.
7. Again, the PDF auto-reloads but now it is possible to scroll into the blank space beyond the first page. The `*eaf*` buffer displays:
```QPainter::end: Painter ended with 2 saved states
Traceback (most recent call last):
  File "/home/pvankoughnett/.emacs.d/elpa/27.2/develop/eaf-20211104.2004/app/pdf-viewer/buffer.py", line 1019, in paintEvent
    qpixmap = self.get_page_pixmap(index, self.scale * hidpi_scale_factor, self.rotation)
  File "/home/pvankoughnett/.emacs.d/elpa/27.2/develop/eaf-20211104.2004/app/pdf-viewer/buffer.py", line 941, in get_page_pixmap
    page = self.document[index]
  File "/home/pvankoughnett/.emacs.d/elpa/27.2/develop/eaf-20211104.2004/app/pdf-viewer/buffer.py", line 398, in __getitem__
    page = PdfPage(self.document[index], index, self.document.isPDF)
  File "/home/pvankoughnett/.local/lib/python3.8/site-packages/fitz/fitz.py", line 5135, in __getitem__
    raise IndexError("page not in document")
IndexError: page not in document
QBackingStore::endPaint() called with active painter; did you forget to destroy it or call QPainter::end() on it?
```

**Expected behavior**
When reloading the file, eaf-pdf-viewer should re-check its page count and display the file appropriately.

**Versions (please complete the following info):**
 - Distro and DE/WM: Ubuntu/Gnome
 - Versions of Dependencies: eaf-20211104.2004
 - M-x emacs-version: GNU Emacs 27.2 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.20, cairo version 1.16.0) of 2021-11-05; using Spacemacs v.0.300.0.

**Comment by nbfields** [comment]
I concur that this error exists. It also causes a further problem - if you reload the PDF with r, this does resolve the page number, but further updates to the PDF (ie if writing a LaTeX document and compiling repeatedly) are not automatically seen in the reloaded PDF unless you scroll away from the changed page and then back again. This means that, if writing a multi-page PDF, you either must continually scroll to and from the changed page, or kill and reopen the buffer for the PDF.

**Comment by manateelazycat** [comment]
@luhuaei Can you look this issue?

**Comment by luhuaei** [comment]
> 6\. Again, the PDF auto-reloads but now it is possible to scroll into the blank space beyond the first page. The `*eaf*` buffer displays:

Look like pdf document reload but current page number not updated, causing this problem.

**Comment by luhuaei** [comment]
Hi, @nbfields  @allispaul Thanks reports. I try to fix this bug on #49

**Comment by luhuaei** [comment]
I close this issue. On my test, repeated modify document already not cause this problem. Re-open this issue, if this problem still in.
